tasks/GenerateReadme.cfc

```diff
@@ -0,0 +1,202 @@
+			}
+			/* Write the file */
+			fileWrite( readMeFile , readmeContent );
+
+			print.greenLine( 'Readme generated!' )
+				.indentedYellowLine( readMeFile )
+				.toConsole();
+		}
+		else {
+			print.redLine("README.md already exists! Use the overwrite=true to overwrite the file.")
+				.indentedYellowLine( readMeFile )
+				.toConsole();
+		}
+
+	}
+
+	/**
+	*
+	* Get the optional .meta/hint.md content
+	*
+	*/
+	private function getHints( required string slug ) {
+
+		var hintsFilePath = repoRootPath & "/exercises/" & arguments.slug & "/.meta/hints.md";
+		var hints = fileExists( hintsFilePath ) ? fileRead( hintsFilePath ) : "";
+
+		return hints;
+	}
+
+	private function getSpec( required string slug ) {
+
+		var spec = getExerciseMetadata( arguments.slug );
+
+		/* Use title from metadata.yml if provided else we must use the slug */
+		if ( len( spec.title )) {
+			spec.name = spec.title;
+		}
+		else {
+			spec.name = titleCaseSlug( arguments.slug );
+		}
+
+		spec.credits = getCredits( spec );
+		spec.description = getDescription( arguments.slug );
+
+		return spec;
+	}
+
+	/**
+	*
+	* Get the exercise description
+	*
+	*/
+	private function getDescription( required string slug ) {
+
+		var descriptionFilePath = problemSpecsPath & "/exercises/" & arguments.slug & "/description.md";
+		 
+		return fileExists( descriptionFilePath ) ? trim( fileRead( descriptionFilePath ) ) : "";
+	}
+
+	/**
+	*
+	* Get the content for exercise credits
+	*
+	*/
+	private function getCredits( required struct spec ) {
+
+		if ( spec.source_url == "" ) {
+			var credits =  spec.source;
+		}
+		else if ( spec.source == "") {
+			var credits =  "[#spec.source_url#](#spec.source_url#)";
+		}
+		else {
+			var credits = "#spec.source# [#spec.source_url#](#spec.source_url#)";
+		}
+
+		return  credits;
+	}
+
+	/**
+	*
+	* Convert exercise slug to a title
+	*
+	*/
+	private function titleCaseSlug( required string slug ) {
+		return trim( reReplace( arguments.slug, "(\b|-)([a-zA-Z])", " \u\2", "all" ) );
+	}
+
+	/**
+	*
+	* Get exercise metadata from metadata.yml in the problem-specifications
+	*
+	*/
+	private struct function getExerciseMetadata( required string slug ) {
+
+		var metadataFilePath = problemSpecsPath & "/exercises/" & arguments.slug & "/metadata.yml";
+		var content = fileRead(metadataFilePath);
+		var metadata = {};
+
+		/* Get the title if provided in metadata.yml */
+		metadata.title = parseMetadata( "title", content );
+
+		/* Get the source of the exercise if provided in metadata.yml */
+		metadata.source = parseMetadata( "source", content );
+
+		/* Get the source url if provided in metadata.yml */
+		metadata.source_url = parseMetadata( "source_url", content );
+
+		return metadata;
+	}
+
+	/**
+	*
+	* Parse metadata from metadata.yml content
+	*
+	*/
+	private string function parseMetadata( required string key, required string content ) {

```

Comment:
It wouldn't be a bad idea to make a module that provides YAML processing.  I wish CFML did this out of the box, but there's some nice drop-in jars that handle it that I've used before.  I wouldn't worry about changing this if it works, just an idea for later.  Right now there's not a super duper good way to "include" arbitrary modules for task runners.  There's a number of ways to manually load up a module in CommandBox,  but I haven't decided on any sort of convention per se for making it seamless.  

Existing YAML lib that would be a good candidate for modularization: 
https://github.com/Ortus-Solutions/vagrant-centos-lucee/blob/master/vagrant/provisioners/lib/YAMLParser.cfc

Reply:
I looked at a couple Java libraries for doing that but was not sure of the best way to use in a task runner.

Reply:
The best way would be to write a general 'Box module that exposes a model that handles YAML parsing like the one I linked to above.  If we wanted it to work on Adobe, we'd need to depend on javaloader as well since Adobe doesn't provide the same features Lucee has for on-the-fly loading of jars.  (`this.javaSettings` was a good idea, but doesn't work in a modular mindset) Creating the module would be the easy part, then we'd have to figure out where and how to get it into the project (probably a `box install` somewhere that maintainers would need to run) and then the task runner would need to load the module up to use it.  This is all possible, I'm just not satisfied with the experience.  See this ticket for more discussion on the matter.
https://ortussolutions.atlassian.net/browse/COMMANDBOX-669